setup.py

```diff
@@ -1,23 +1,10 @@
 from setuptools import setup, find_packages
-from subprocess import check_output
-
-
-def get_version():
-    try:
-        tag = check_output(
-            ["git", "describe", "--tags", "--abbrev=0", "--match=[0-9]*"]
-        )
-        return tag.strip("\n")
-    except Exception:
-        # if somehow you get the repo not from git,
-        # hardcode default major version
-        return "2.0.0"
 
 
 setup(
     name="dictionaryutils",
-    version=get_version(),
+    version='2.0.5',
```

Comment:
Are we certain we want to do this? Removing automation so we can get out of sync again?

Reply:
Thanks for the review!

https://pypi.org/project/dictionaryutils/2.0.4/ - the "automation" is supposed to be falling back to `2.0.4` if not git-controlled, but it is actually `2.0.0`. This bug could be fixed, but in the long run it is not wise to mess up with the versions in `setup.py`, because [Explicit is better than implicit](https://www.python.org/dev/peps/pep-0020/).

For sure automation is preferred, we just need a better approach for all packages. ([bumpversion](https://github.com/peritus/bumpversion) for example, you're more than welcome to recommend different solutions.)

Reply:
None of this will make our life easier... keeping the GitHub tags/releases in sync with the PyPi version is a dramatic quality of life improvement though.

For the bug, the deploy section of .travis.yml is missing a skip_cleanup: true

Reply:
I apologize if I was not clear about:

> For sure automation is preferred, we just need a better approach for all packages.

I meant exactly "keeping the GitHub tags/releases in sync with the PyPi version".

Reply:
Okay, but you're replacing working automation with no automation until we find another working solution we think is "better"? Lets just keep the existing for now...

Reply:
I think we should now start using [bumpversion](https://github.com/peritus/bumpversion) to create git tags **and** bumping versions in setup.py, after my PRs syncing versions in library repos. Having **one** repo working inconsistently is a bit weird.

The bumpversion process will not make life harder, sorry I should have shown you this first:

1. Pull up to latest in git (or releasing version).
2. Run `bumpversion patch` or `bumpversion minor`. It will:
    1. Update version in `setup.py`.
    2. Commit the changes in git.
    3. Create a git tag with the new version.
3. Push the commit with the tag.

It has the same number of steps as current process with `dictionaryutils`, only replacing `git tag 2.1.8` (or `gen3git tag 2.1.8`) with `bumpversion patch`, and you don't even need to calculate the version numbers by yourself.

Regarding the `gen3git` [CLI tool](https://github.com/uc-cdis/release-helper), I can have bumpversion integrated, so that `gen3git bumpversion` command could create git tags with annotation with release notes.

Still, this is only my proposal, please let me know if any issue and feel free to suggest a different one. But overall I think repositories should have a consistent way managing versions/tags.

Reply:
Thats 3 times as many steps as before... our goal is to make life as easy as possible for devs. We should be using this model for all the repos because its what we use for the microservices too. You create a new git release/tag when you want to have a version to pin to in a deployment. Why not just keep this same process for python code instead of having a different process for different things?

Reply:
> Thats 3 times as many steps as before

Sorry I might misunderstood what steps are before. I thought they were:

1. 1. Pull up to latest in git (or releasing version).
2. Run `git tag 2.1.8` or `gen3git tag 2.1.8`.
3. Push the commit with the tag.

As far as we are all clear about the facts of the two approaches, I have no preference for personal taste on one over another. Please make the call and I'll get back to all the repos to fix them, maybe a bit later after the sheepdog fixes.

Reply:
You can do all the git steps within the github.com website, so theres not even a need for command line making it super easy without the need for extra python tools, etc. And then the process is exactly the same everywhere.

We should use the git tags/release process for setting the versions.

Reply:
Ah okay! Thanks for the clarification and sorry for the lack of information. I'll try to use Travis to run bumpversion when triggered by git tag, and let you know the comparison results a bit later. Thanks for working with me!